engine/crates/engine-v2/src/engine.rs

```diff
@@ -9,20 +11,24 @@ use crate::{
 };
 
 pub struct Engine {
-    pub(crate) schema: Schema,
+    // We use an Arc for the schema to have a self-contained response which may still
+    // needs access to the schema strings
```

Comment:
100% agree with this move. Not saying that we have to deal with it right now, but have you thought about how to reload the schema on a running engine? (it might be as simple as `Arc<RwLock<Schema>>`)

Reply:
IMO a new engine should be built, it might need new services for connectors (ex: connection pooling) and the query plan cache shouldn't be re-used after schema changes. In my mind, the engine is only responsible for executing a request given a certain schema, it's immutable.

Reply:
Yeah the reloading doesn't need to be at this level, the HTTP server can start a whole new engine, fair point.